c2corg_ui/static/js/search.js

```diff
@@ -163,44 +172,58 @@ app.SearchController = function($scope, $compile, $attrs, apiUrl, gettextCatalog
 
 /**
  * @param {string} type The document type.
- * @param {string} title The category title.
  * @return {TypeaheadDataset} A data set.
  * @private
  */
-app.SearchController.prototype.createDataset_ = function(type, title) {
+app.SearchController.prototype.createDataset_ = function(type) {
   var bloodhoundEngine = this.createAndInitBloodhound_(type);
   return /** @type {TypeaheadDataset} */({
+    contents : type,
     source: bloodhoundEngine.ttAdapter(),
     display: function(doc) {
       return doc.label;
     },
     limit: 20,
     templates: {
       header: (function() {
-        return '<div class="header ' + title + '">' +
-            this.gettextCatalog_.getString(title) + '</div>';
+        return '<div class="header" dataset="' + type + '">' + this.gettextCatalog_.getString(type) + '</div>';
       }).bind(this),
+      footer: function() {
+        return '<p class="suggestion-more"><a href="/' + type + '" class="green-text">+ see more results</a></p>';
+      },
       suggestion: function(doc) {
-        return '<p>' + doc.label + '</p>';
-      }
-      // Note: The templates for `notFound` and `pending` are not set, because
-      // they would be shown for each dataset.
+        var document = {
+          'query' : doc['_query'],
+          'elevation' : doc['elevation'] || doc['elevation_max'],
+          'waypoint_type' : doc['waypoint_type'],
+          'activities' : doc['activities'],
+          'date_end' : doc['date_end'],
+          'doctype' : doc['documentType']
+        };
```

Comment:
Why not directly passing ``doc`` to the ``app-suggestion`` directive instead of creating this intermediary ``document`` variable? Then we would leave the templates use the attributes they need.

Depending on the document type you may have really different attributes. ``waypoint_type`` is only relevant for waypoints. ``date_end`` only for outings.